Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/mysql/MySQLNode.java

```diff
@@ -0,0 +1,178 @@
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.tooling.msservices.serviceexplorer.azure.mysql;
+
+import com.microsoft.azure.management.mysql.v2020_01_01.Server;
+import com.microsoft.azure.management.mysql.v2020_01_01.ServerState;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTask;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import com.microsoft.azuretools.core.mvp.model.mysql.MySQLMvpModel;
+import com.microsoft.azuretools.telemetry.AppInsightsConstants;
+import com.microsoft.azuretools.telemetry.TelemetryProperties;
+import com.microsoft.tooling.msservices.helpers.Name;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureRefreshableNode;
+import com.microsoft.tooling.msservices.serviceexplorer.Node;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeAction;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.AzureNodeActionPromptListener;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLModule.ICON_FILE;
+
+
+public class MySQLNode extends Node implements TelemetryProperties {
+
+    @Getter
+    private final String subscriptionId;
+    @Getter
+    private final Server server;
+    private ServerState serverState;
+
+    public MySQLNode(AzureRefreshableNode parent, String subscriptionId, Server mysqlServer) {
+        super(mysqlServer.id(), mysqlServer.name(), parent, ICON_FILE, true);
+        this.subscriptionId = subscriptionId;
+        this.server = mysqlServer;
+        this.serverState = server.userVisibleState();
+        loadActions();
+    }
+
+    public Map<String, String> toProperties() {
+        final Map<String, String> properties = new HashMap<>();
+        properties.put(AppInsightsConstants.SubscriptionId, this.subscriptionId);
+        // TODO: track region name
+        return properties;
+    }
+
+    @Override
+    protected void loadActions() {
+        initActions();
+    }
+
+    @Override
+    public List<NodeAction> getNodeActions() {
+        boolean running = ServerState.READY.equals(serverState);
+        getNodeActionByName(StartAzureMySQLAction.ACTION_NAME).setEnabled(!running);
+        getNodeActionByName(StopAzureMySQLAction.ACTION_NAME).setEnabled(running);
+        getNodeActionByName(RestartAzureMySQLAction.ACTION_NAME).setEnabled(running);
+        return super.getNodeActions();
+    }
+
+    private void refreshNode() {
+        Server result = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
```

Comment:
handle `result = null`?

Reply:
it's ok in a private method. and "findServer" method will throw exception if server doesn't exist.